pkg/yurtadm/cmd/renew/certificate/certificate.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2023 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certificate
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	"k8s.io/klog/v2"
+
+	"github.com/openyurtio/openyurt/pkg/projectinfo"
+	yurtconstants "github.com/openyurtio/openyurt/pkg/yurtadm/constants"
+	"github.com/openyurtio/openyurt/pkg/yurtadm/util/yurthub"
+)
+
+type certificateOptions struct {
+	token                    string
+	caCertHashes             []string
+	unsafeSkipCAVerification bool
+	serverAddr               string
+	yurthubServer            string
+}
+
+// newCertificateOptions returns a struct ready for being used for creating cmd renew flags.
+func newCertificateOptions() *certificateOptions {
+	return &certificateOptions{
+		caCertHashes:             make([]string, 0),
+		unsafeSkipCAVerification: true,
+		yurthubServer:            yurtconstants.DefaultYurtHubServerAddr,
+	}
+}
+
+// NewCmdCertificate returns "yurtadm renew certificate" command.
+func NewCmdCertificate() *cobra.Command {
+	o := newCertificateOptions()
+
+	certificateCmd := &cobra.Command{
+		Use:   "certificate",
+		Short: "Create bootstrap file for yurthub to update certificate",
+		RunE: func(certificateCmd *cobra.Command, args []string) error {
+			if err := o.validate(); err != nil {
+				klog.Fatalf("validate options: %v", err)
+			}
+
```

Comment:
I think we should verify the certificates of Yurthub expired or not at first. and if the certificate is valid, we can exit `yurtadm renew certificate` directly with some logs.

Reply:
Fixed